subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/internal/CheckstyleInvoker.groovy

```diff
@@ -77,11 +77,7 @@ abstract class CheckstyleInvoker {
                     // User provided their own config_loc
                     def userProvidedConfigLoc = configProperties[CONFIG_LOC_PROPERTY]
                     if (userProvidedConfigLoc) {
-                        DeprecationLogger.deprecateIndirectUsage("Adding 'config_loc' to checkstyle.configProperties")
-                            .withAdvice("This property is now ignored and the value of configDirectory is always used for 'config_loc'.")
-                            .willBeRemovedInGradle7()
-                            .withUpgradeGuideSection(5, "user_provided_config_loc_properties_are_ignored_by_checkstyle")
-                            .nagUser()
+                        throw new InvalidUserDataException("Cannot add config_loc property checkstyle.configProperties")
```

Comment:
I think it would be good to keep the information to use `configDirectory` in the error message. Something like:

`Please configure the configDirectory on the checkstyle task instead`

The upgrade guide could also mention that.

Reply:
Fixed